src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenInParametersTests.cs

```diff
@@ -1165,6 +1182,21 @@ public void M1(in int arg2, in int arg3, in int arg4)
 42
 3
 3");
+
+            comp = CreateCompilationWithMscorlib46(text, new[] { ValueTupleRef, SystemRuntimeFacadeRef }, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithPEVerifyCompatFeature());
+            CompileAndVerify(comp, verify: false, expectedOutput: @"
+3
+42
+2
```

Comment:
The output is different because the flag currently deoptimizes overal treatment of readonly fields as lvalues in readonly context. Thus it affects scenarios where removing aliasing is observable. Generally those are scenarios involving `readonly ref` 

This makes me think that perhaps we should not deoptimize those. 
It could be hard, however, to distinguish situations where new features are involved since that could be very indirect.